File: tests/machine/x/zig/match_full.zig
const std = @import("std");

const x = 2; // i32
const label = switch (x) {1 => "one", 2 => "two", 3 => "three", else => "unknown", }; // []const u8
const day = "sun"; // []const u8
const mood = switch (day) {"mon" => "tired", "fri" => "excited", "sun" => "relaxed", else => "normal", }; // []const u8
const ok = true; // bool
const status = switch (ok) {true => "confirmed", false => "denied", }; // []const u8

fn classify(n: i32) []const u8 {
    return switch (n) {0 => "zero", 1 => "one", else => "many", };
}

pub fn main() void {
    std.debug.print("{s}\n", .{label});
    std.debug.print("{s}\n", .{mood});
    std.debug.print("{s}\n", .{status});
    std.debug.print("{s}\n", .{classify(0)});
    std.debug.print("{s}\n", .{classify(5)});
}

File: tests/machine/x/lua/match_full.lua
function __print(...)
    local args = {...}
    local parts = {}
    for i,a in ipairs(args) do
        if a ~= nil and a ~= '' then parts[#parts+1] = tostring(a) end
    end
    print(table.concat(parts, ' '))
end
function classify(n)
  return (function()
  local _t0 = n
  if _t0 == 0 then return "zero" end
  if _t0 == 1 then return "one" end
  return "many"
end)()
end

x = 2
label = (function()
  local _t1 = x
  if _t1 == 1 then return "one" end
  if _t1 == 2 then return "two" end
  if _t1 == 3 then return "three" end
  return "unknown"
end)()
__print(label)
day = "sun"
mood = (function()
  local _t2 = day
  if _t2 == "mon" then return "tired" end
  if _t2 == "fri" then return "excited" end
  if _t2 == "sun" then return "relaxed" end
  return "normal"
end)()
__print(mood)
ok = true
status = (function()
  local _t3 = ok
  if _t3 == true then return "confirmed" end
  if _t3 == false then return "denied" end
  return nil
end)()
__print(status)
__print(classify(0))
__print(classify(5))
